src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -173,9 +176,28 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                     functionName = string.IsNullOrEmpty(fex.MethodName) ? string.Empty : fex.MethodName.Replace("Host.Functions.", string.Empty);
                 }
 
-                (innerExceptionType, innerExceptionMessage, details) = exception.GetExceptionDetails();
-                formattedMessage = Sanitizer.Sanitize(formattedMessage);
-                innerExceptionMessage = innerExceptionMessage ?? string.Empty;
+                var baseEx = exception.GetBaseException();
+                innerExceptionType = baseEx.GetType().FullName;
+
+                var originalMessage = baseEx.Message;
+                var formattedDetails = exception.ToFormattedString();
+
+                const string RedactedMessage = "An exception occurred during invocation, but its details are redacted. Customers with AppInsights or OTel enabled can access full exception details.";
+
+                static string BuildReplacement(string msg)
```

Comment:
Should we consider moving `BuildReplacement()` method outside instead of nested method, since the `Log()` method is already quite large?

Some naming ideas for `BuildReplacement()` method - 

`BuildReplacementExceptionMessage()`
`BuildRedactedExceptionMessage()`
`GetRedactedExceptionMessage()`